theano/compile/builders.py

```diff
@@ -55,14 +61,13 @@ class OpFromGraph(gof.Op):
           local_outputs)
         - c_code() to remove the double overhead?
         - grad() make it support DisconnectedType and the new interface
-        - implement R_op()
         - check how it works with updates.
         - add test with constant as input or inside the inner graph.
         - Add support for the GPU? Probably just need an opt to remove transfer
         - Add support to pickle this Op.
         - Add support/test with random generator
-        - Recursion detection to prevent Op "forkbomb", either set depth
-          limit or manually check them.
+        - Add optimization prior to inilne expansion such as removing unused
```

Comment:
inilne -> inline

But I'm not sure I understand everything. What do you mean by "inline expansion"? I understand that we still miss opt that will remove useless inputs/outputs.